python/lsst/scarlet/lite/parameters.py

```diff
@@ -0,0 +1,463 @@
+    return phi, psi
+
+
+phi_psi = {
+    "adam": _adam_phi_psi,
+    "nadam": _nadam_phi_psi,
+    "amsgrad": _amsgrad_phi_psi,
+    "padam": _padam_phi_psi,
+    "adamx": _adamx_phi_psi,
+    "radam": _radam_phi_psi,
+}
+
+
+class SingleItemArray:
+    """Mock an array with only a single item"""
+
+    def __init__(self, value):
+        self.value = value
+
+    def __getitem__(self, item):
+        return self.value
+
+
+class AdaproxParameter(Parameter):
+    """Operator updated using te Proximal ADAM algorithm
+
+    Uses multiple variants of adaptive quasi-Newton gradient descent
+        * Adam (Kingma & Ba 2015)
+        * NAdam (Dozat 2016)
+        * AMSGrad (Reddi, Kale & Kumar 2018)
+        * PAdam (Chen & Gu 2018)
+        * AdamX (Phuong & Phong 2019)
+        * RAdam (Liu et al. 2019)
+    See details of the algorithms in the respective papers.
+    """
+
+    def __init__(
+        self,
+        x: np.ndarray,
+        step: Callable | float,
+        grad: Callable | None = None,
+        prox: Callable | None = None,
+        b1: float = 0.9,
+        b2: float = 0.999,
+        eps: float = 1e-8,
+        p: float = 0.25,
+        m0: np.ndarray | None = None,
+        v0: np.ndarray | None = None,
+        vhat0: np.ndarray | None = None,
+        scheme: str = "amsgrad",
+        prox_e_rel: float = 1e-6,
+    ):
+        shape = x.shape
+        dtype = x.dtype
+        if m0 is None:
+            m0 = np.zeros(shape, dtype=dtype)
+
+        if v0 is None:
+            v0 = np.zeros(shape, dtype=dtype)
+
+        if vhat0 is None:
+            vhat0 = np.ones(shape, dtype=dtype) * -np.inf
+
+        super().__init__(
+            x,
+            {
+                "m": m0,
+                "v": v0,
+                "vhat": vhat0,
+            },
+            step,
+            grad,
+            prox,
+        )
+
+        if isinstance(b1, float):
+            _b1 = SingleItemArray(b1)
+        else:
+            _b1 = b1
+
+        self.b1 = _b1
+        self.b2 = b2
+        self.eps = eps
+        self.p = p
+
+        self.phi_psi = phi_psi[scheme]
+        self.e_rel = prox_e_rel
+
+    def update(self, it: int, input_grad: np.ndarray, *args):
+        """Update the parameter and meta-parameters using the PGM
+
+        See `~LiteParameter` for more.
+        """
+        _x = self.x
+        # Calculate the gradient
+        grad = cast(Callable, self.grad)(input_grad, _x, *args)
+        # Get the update for the parameter
+        phi, psi = self.phi_psi(
+            it,
+            grad,
+            self.helpers["m"],
+            self.helpers["v"],
+            self.helpers["vhat"],
+            self.b1,
+            self.b2,
+            self.eps,
+            self.p,
+        )
+        # Calculate the step size
+        step = self.step
+        if it > 0:
+            _x += -step * phi / psi
+        else:
+            # This is a scheme that Peter Melchior and I came up with to
```

Comment:
No, I don't think there was ever a ticket for this. It was something we included in one of our rebases a few years ago because it just worked through trial and error. There is probably a conversation somewhere in our slack DM together (or possibly including Remy) but nothing more official than that.